File: convert-binary-number-in-a-linked-list-to-integer/convert-binary-number-in-a-linked-list-to-integer.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        '''
        Given a linked list representing a binary number
        Returns the value of that binary number as an int
        '''
        num = 0
        while head is not None:
            if head.val == 1:
                num *= 2
                num += 1
            else:
                num *= 2
            head = head.next
        return num

File: convert-binary-number-in-a-linked-list-to-integer/convert-binary-number-in-a-linked-list-to-integer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution 
{
    public int getDecimalValue(ListNode head) 
    {
        // Iterate thru the list using bitwise or to append binary
        int answer = 0;
        while (head != null)
        {
            answer = answer << 1;
            answer = answer ^ head.val;
            head = head.next;
        }
        
        return answer;
    }
}